from snakemake.utils import min_version

##### set minimum snakemake version #####

min_version("6.4.1")


##### setup config and necessary parameters #####


configfile: "config/config.yaml"


##### setup report #####


report: "report/workflow.rst"


##### setup singularity #####


# this container defines the underlying OS for each job when using the workflow
# with --use-conda --use-singularity
container: "docker://continuumio/miniconda3"


##### include rules #####
include: "rules/ref.smk"
include: "rules/spike.smk"
include: "rules/flatten.smk"
include: "rules/combine.smk"




##### target rules #####

if(config["spike_in"]):

    rule all:
        input:
            "results/combined/combined_genome.gtf",
            "results/flattened/flat_genome_exonID.gtf",
            "results/combined/combined_genome.fasta"

else:

    rule all:
        input:
            "results/flattened/flat_genome_exonID.gtf",
            "results/genome/genome.fasta"